kyo-core/shared/src/main/scala/kyo/Adder.scala

```diff
@@ -0,0 +1,25 @@
+package kyo
+
+import java.util.concurrent.atomic as j
+
+class LongAdder private[kyo] (private val ref: j.LongAdder) extends AnyVal:
+    inline def add(v: Long)(using Frame): Unit < IO = IO(ref.add(v))
+    inline def decrement(using Frame): Unit < IO    = IO(ref.decrement())
+    inline def increment(using Frame): Unit < IO    = IO(ref.increment())
+    inline def get(using Frame): Long < IO          = IO(ref.sum())
+    inline def reset(using Frame): Unit < IO        = IO(ref.reset())
+    inline def sumThenReset(using Frame): Long < IO = IO(ref.sumThenReset())
+end LongAdder
+
+object LongAdder:
+    def init(using Frame): LongAdder < IO = IO(LongAdder(new j.LongAdder))
```

Comment:
I'm not sure I understood, can you elaborate more?

Reply:
The comment is generally irrelevant to the line.

It's preferable to avoid expanding `Frame`s in Kyo's codebase, as the goal is to find where the methods are called in user code. For this reason, it may be useful to disable expanding the `Frame` to a value in Kyo's code.

ZIO does so with an implicit 'marker' class:
https://github.com/zio/zio/blob/series/2.x/stacktracer/shared/src/main/scala/zio/stacktracer/TracingImplicits.scala
https://github.com/zio/zio/blob/series/2.x/stacktracer/shared/src/main/scala-3/zio/internal/stacktracer/Macros.scala#L51-L67

Reply:
Thanks for the clarification! I agree it'd be good to have this restriction mechanism. Maybe we could have some more strict in the macro to always fail to compile in case the frame is in a symbol under the `kyo` package?

Reply:
That would be great, but we may also need the flag for library authors at some point.